File: backend/internal/dto/response/deposit.go
package response

import "time"

type DepositAddressResp struct {
	Address  string `json:"address"`
	Currency string `json:"currency"`
	Network  string `json:"network"`
}

type DepositOrderResp struct {
	ID        string    `json:"id"`
	Currency  string    `json:"currency"`
	Network   string    `json:"network"`
	Amount    string    `json:"amount"`
	TxHash    *string   `json:"tx_hash,omitempty"`
	Status    string    `json:"status"`
	CreatedAt time.Time `json:"created_at"`
}

type DepositOrderListResp struct {
	Orders []DepositOrderResp `json:"orders"`
	Total  int64              `json:"total"`
}

File: admin/src/types/deposit.ts
export interface AdminDeposit {
  id: number;
  platform_order_id: string;
  merchant_id: number;
  merchant_email: string;
  currency: string;
  network: string;
  amount: string;
  tx_hash?: string;
  to_address: string;
  from_address?: string;
  status: string;
  created_at: string;
  completed_at?: string;
}

export interface AdminDepositListResponse {
  deposits: AdminDeposit[];
  total: number;
}
